events/janus_gelfevh.c

```diff
@@ -0,0 +1,476 @@
+		if(json_object_get(request, "port"))
+			req_port = json_string_value(json_object_get(request, "port"));
+		if(req_backend && req_port) {
+			/* Invalid backend address, port */
+			error_code = JANUS_GELFEVH_ERROR_INVALID_ELEMENT;
+			g_snprintf(error_cause, sizeof(error_cause), "Invalid backend URI '%s:%s'", req_backend, req_port);
+			goto plugin_response;
+		}
+		/* If we got here, we can enforce */
+		if(req_events)
+			janus_events_edit_events_mask(req_events, &janus_gelfevh.events_mask);
+		if(req_backend) {
+			janus_mutex_lock(&evh_mutex);
+			if(req_backend && req_port) {
+				g_free(backend);
+				g_free(port);
+				backend = g_strdup(req_backend);
+				port = g_strdup(req_port);
+			}
+			janus_mutex_unlock(&evh_mutex);
+		}
+	} else {
+		JANUS_LOG(LOG_VERB, "Unknown request '%s'\n", request_text);
+		error_code = JANUS_GELFEVH_ERROR_INVALID_REQUEST;
+		g_snprintf(error_cause, 512, "Unknown request '%s'", request_text);
+	}
+
+plugin_response:
+		{
+			json_t *response = json_object();
+			if(error_code == 0) {
+				/* Return a success */
+				json_object_set_new(response, "result", json_integer(200));
+			} else {
+				/* Prepare JSON error event */
+				json_object_set_new(response, "error_code", json_integer(error_code));
+				json_object_set_new(response, "error", json_string(error_cause));
+			}
+			return response;
+		}
+}
+
+/* Thread to handle incoming events */
+static void *janus_gelfevh_handler(void *data) {
+	JANUS_LOG(LOG_VERB, "Joining GelfEventHandler handler thread\n");
+	json_t *event = NULL, *output = NULL;
+	static char *event_text = NULL;
+	const char *short_message = NULL;
+
+	while(g_atomic_int_get(&initialized) && !g_atomic_int_get(&stopping)) {
+		event = g_async_queue_pop(events);
+		if(event == NULL)
+			continue;
+		if(event == &exit_event)
+			break;
+		output = NULL;
+
+		while(TRUE) {
+			/* Handle event */
+			int type = json_integer_value(json_object_get(event, "type"));
+			switch(type) {
+				case JANUS_EVENT_TYPE_SESSION:
+					short_message = "JANUS_EVENT_TYPE_SESSION";
+					break;
+				case JANUS_EVENT_TYPE_HANDLE:
+					short_message = "JANUS_EVENT_TYPE_HANDLE";
+					break;
+				case JANUS_EVENT_TYPE_JSEP:
+					short_message = "JANUS_EVENT_TYPE_JSEP";
+					break;
+				case JANUS_EVENT_TYPE_WEBRTC:
+					short_message = "JANUS_EVENT_TYPE_WEBRTC";
+					break;
+				case JANUS_EVENT_TYPE_MEDIA:
+					short_message = "JANUS_EVENT_TYPE_MEDIA";
+					break;
+				case JANUS_EVENT_TYPE_PLUGIN:
+					short_message = "JANUS_EVENT_TYPE_PLUGIN";
+					break;
+				case JANUS_EVENT_TYPE_TRANSPORT:
+					short_message = "JANUS_EVENT_TYPE_TRANSPORT";
+					break;
+				case JANUS_EVENT_TYPE_CORE:
+				case JANUS_EVENT_TYPE_EXTERNAL:
+					short_message = "JANUS_EVENT_TYPE_CORE";
+					break;
+				default:
+					JANUS_LOG(LOG_WARN, "Unknown type of event '%d'\n", type);
+					short_message = "UNKNOWN_JANUS_EVENT";
+					break;
+			}
+			output = event;
```

Comment:
This is going to cause issues, since you're modifying the event, and then extracting portions of the original event and setting them on itself. Just create a new object here (`output = json_object()`), and copy what you need from the original event if this needs to be in a format Graylog expects.

Reply:
This is quite important, so please make sure this is addressed in the next commits.

Reply:
Done, thanks for a great find